### PR TITLE
Delete website_docs since it has moved

### PR DESCRIPTION
**Pull request by chalin**
- Delete `website_docs` because, as of https://github.com/open-telemetry/opentelemetry.io/pull/995, the OTel website pulls Java docs from https://github.com/open-telemetry/opentelemetry-java-docs
- Update references to the deleted directory

/cc @austinlparker @trask

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -47,7 +47,9 @@ Next, update the
 
 Create a PR to mark the new release in README.md and CHANGELOG.md on the main branch.
 
-Finally, update the files `website_docs` directory to point at the newly released version.
+Finally, update the the [website docs][] to refer to the newly released version.
```
Comment (trask) [inline RELEASING.md:50]:
```suggestion
Finally, update the the [website docs] to refer to the newly released version.
```

**Review by trask — APPROVED** [review]

File: RELEASING.md (hunk 1 above)
Comment (chalin) [reply RELEASING.md:50]: By using a trailing `[]` on markdown link references, editors like VS Code give you visual feedback if your link is defined or not. So I'd suggest keeping it, but it's your call.

Comment (chalin) [reply RELEASING.md:50]: (This is _really_ useful as-you-type feedback. :))

Comment (trask) [reply RELEASING.md:50]: does it makes sense to add it above also on the `[Manual instrumentation]` link?

(also just noticed an extra `the`)

```suggestion
Finally, update the [website docs][] to refer to the newly released version.
```

Comment (chalin) [reply RELEASING.md]: Thanks for catching the stutter.

Comment (chalin) [reply RELEASING.md]: > does it makes sense to add it above also on the `[Manual instrumentation]` link?

Done

**Review by kubawach — APPROVED** [review]
LGTM!

**Review by anuraaga — APPROVED** [review]

**Pull request merged**